### DISCUSSION TITLE
Blueprint Action feature

### DISCUSSION THREAD

**Discussion by DerSpeedMann** [SmartlyDressedGames/Unturned-3.x-Community#3639] [Ideas]
The Blueprint Action should prioritize the Item on which the Blueprint Action was called.
Or at least send InventorGoup and index or ItemJar of the item with the onCraftBlueprintRequested or max value / null if crafting was called from the crafting menu.

**Comment by liebesleid-eth** [comment]
what is this for

**Reply by DerSpeedMann** [reply]
If you right click an item and use any of the blueprint actions, it does not apply to the selected item.
It instead just invokes the normal crafting logic, which selects lowest quality / first in inventory  

This would also allow for custom plugin actions

**Reply by liebesleid-eth** [reply]
Ohh I see, so thats why sometimes the 100% quality item gets shredded instead of the one you selected XD

**Reply by DerSpeedMann** [reply]
Yes exactly
